query/src/test/java/org/infinispan/query/projection/ScoreProjectionTest.java

```diff
@@ -0,0 +1,104 @@
+
+@Test(groups = "functional", testName = "query.projection.ScoreProjectionTest")
+public class ScoreProjectionTest extends SingleCacheManagerTest {
+
+   @Override
+   protected EmbeddedCacheManager createCacheManager() throws Exception {
+      ConfigurationBuilder indexed = new ConfigurationBuilder();
+      indexed.indexing().enable()
+            .storage(LOCAL_HEAP)
+            .addIndexedEntity(Game.class);
+
+      return TestCacheManagerFactory.createCacheManager(indexed);
+   }
+
+   @BeforeMethod
+   public void setUp() {
+      if (!cache.isEmpty()) {
+         return;
+      }
+
+      cache.put(4, new Game("4", "descriAAion"));
+      cache.put(1, new Game("1", "description"));
+      cache.put(3, new Game("3", "desription"));
+      cache.put(5, new Game("5", "very-different"));
+      cache.put(2, new Game("2", "descriqtion"));
+   }
+
+   @Test
+   public void entityAndScore() {
+      Query<Object[]> query;
+      List<Object[]> games;
+
+      query = cache.query("select g, score(g) from org.infinispan.query.model.Game g where g.description : 'description'~2");
+      games = query.list();
+      assertThat(games).extracting(objects -> objects[0]).extracting("name").containsExactly("1", "2", "3", "4");
+      assertThat(games).extracting(objects -> objects[1]).hasOnlyElementsOfType(Float.class).isNotNull();
+   }
+
+   @Test
+   public void fieldAndScore() {
+      Query<Object[]> query;
+      List<Object[]> games;
+
+      query = cache.query("select score(g), g.name from org.infinispan.query.model.Game g where g.description : 'description'~2");
+      games = query.list();
+      assertThat(games).extracting(objects -> objects[1]).containsExactly("1", "2", "3", "4");
+      assertThat(games).extracting(objects -> objects[0]).hasOnlyElementsOfType(Float.class).isNotNull();
+   }
+
+   @Test
+   public void entityScoreAndField() {
+      Query<Object[]> query;
+      List<Object[]> games;
+
+      query = cache.query("select g.name, score(g), g from org.infinispan.query.model.Game g where g.description : 'description'~2");
+      games = query.list();
+      assertThat(games).extracting(objects -> objects[0]).containsExactly("1", "2", "3", "4");
+      assertThat(games).extracting(objects -> objects[1]).hasOnlyElementsOfType(Float.class).isNotNull();
+      assertThat(games).extracting(objects -> objects[2]).extracting("name").containsExactly("1", "2", "3", "4");
+   }
+
+   @Test
+   public void nativeQuery() {
+      ComponentRegistry componentRegistry = ComponentRegistry.of(cache);
+      SearchMapping searchMapping = componentRegistry.getComponent(SearchMapping.class);
+      assertThat(searchMapping).isNotNull();
+
+      SearchSession mappingSession = searchMapping.getMappingSession();
+      SearchScope<Game> scope = mappingSession.scope(Game.class);
+
+      SearchProjection<List<?>> projection = scope.projection().composite(scope.projection().score(), scope.projection().entity()).toProjection();
```

Comment:
Just wanted to share that not so long ago, Hibernate Search introduced a way to bind this type of projections like:
```java
@ProjectionConstructor
public record MyProjection(@ScoreProjection float score, @EntityProjection Game entity) {
}
// and then
SearchProjection<MyProjection> projection = scope.projection().composite().as( MyProjection.class ).toProjection();
```
Not that it is needed here, but maybe you'll find it useful in some future tests 😃 
See https://docs.jboss.org/hibernate/stable/search/reference/en-US/html_single/#search-dsl-projection-entity-mapping

Reply:
Thank you @marko-bekhta nice API. Thanks for sharing it!